cmake/CableCompilerSettings.cmake

```diff
@@ -87,6 +87,12 @@ macro(cable_configure_compiler)
             # Enable basing warnings set and treat them as errors.
             add_compile_options(-Werror -Wall -Wextra -Wshadow)
 
+            add_compile_options(-Wdouble-promotion -Wcast-qual -Wcast-align -Wdangling-else -Wlogical-not-parentheses -Wmissing-declarations)
```

Comment:
I think this must be checked against documentation.
E.g. `-Wlogical-not-parentheses`
- is enabled in Clang by default: https://clang.llvm.org/docs/DiagnosticsReference.html#wlogical-not-parentheses
- is enabled in GCC with `-Wall`: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wall

Reply:
I added these based on documentation: first added a lot of the things I liked, then removed those which are included in `-Wall` (and other settings we have). Seems like this one slipped through the cracks.

Reply:
I picked this one randomly.

The `-Wdouble-promotion` should be remove as I don't want the change https://github.com/wasmx/fizzy/pull/508#discussion_r478300508.

I will check the rest.

Reply:
- It seems `-Wcast-qual` is only for C-style casts of pointers. Unlikely we will see such thing in our code, but the warning option can stay.

Reply:
> The -Wdouble-promotion should be remove as I don't want the change #508 (comment).

Do you have a reasoning? I'd be also interested to hear @gumb0's view.

Reply:
@chfast @gumb0 any other warnings we should/could enable?

Reply:
> > The -Wdouble-promotion should be remove as I don't want the change #508 (comment).
> 
> Do you have a reasoning? I'd be also interested to hear @gumb0's view.

The `double{x}` is superior to `static_cast<double>(x)` and the `-Wdouble-promotion` forbids using the first syntax. It is rather a compiler bug due to fact that `-Wdouble-promotion` predates the `double{x}` syntax.

There is even Clang bug for this: https://bugs.llvm.org/show_bug.cgi?id=34062.

Reply:
> @chfast @gumb0 any other warnings we should/could enable?

Yes, try
```
-Weverything -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-padded -Wno-switch-enum -Wno-float-equal -Wno-double-promotion
```
I found many easy fixes with these already.

Reply:
`-Wlogical-not-parentheses` looks to be included in `-Wall`?

Reply:
Move the changes to the main CMakeLists.txt, do not modify the Cable file.

Reply:
This is cleaned up now.